static/app/views/nav/prefersStackedNav.tsx

```diff
@@ -1,12 +1,35 @@
 import ConfigStore from 'sentry/stores/configStore';
+import type {Organization} from 'sentry/types/organization';
+import useOrganization from 'sentry/utils/useOrganization';
 import {useUser} from 'sentry/utils/useUser';
 
-export function prefersStackedNav() {
-  return ConfigStore.get('user')?.options?.prefersStackedNavigation ?? false;
+// IMPORTANT:
+// These two functions NEED to have the same logic. Make sure to update both if you are changing this logic.
+
+export function prefersStackedNav(organization: Organization) {
+  const userStackedNavOption = ConfigStore.get('user')?.options?.prefersStackedNavigation;
+
+  if (
+    userStackedNavOption !== false &&
+    organization.features.includes('enforce-stacked-navigation')
+  ) {
+    return true;
+  }
+
+  return userStackedNavOption ?? false;
 }
 
 export function usePrefersStackedNav() {
   const user = useUser();
+  const organization = useOrganization();
```

Comment:
It might be good to use `{allowNull: true` here since I see that we do that in the top-level org layout: https://github.com/getsentry/sentry/blob/bacea3c92049caa8766158b3a4553fd5bd5d3a53/static/app/views/organizationLayout/index.tsx#L47-L49